File: backend/src/main/java/com/library/backend/entity/Book.java
package com.library.backend.entity;

import com.library.backend.entity.enums.BookStatus;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "books")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String author;

    private String genre;

    @Column(nullable = false, unique = true)
    private String isbn;

    @Column(nullable = false)
    private Integer quantity;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BookStatus status;
}

File: frontend/src/types/book.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  genre: string;
  isbn: string;
  quantity: number;
  status: string;
  description?: string | null;
  summary?: string | null;
  publishedYear?: number | null;
  coverColor?: string | null;
}
